server/src/graql/query/pattern/property/ValueProperty.java

```diff
@@ -119,9 +116,12 @@ public T value() {
         }
 
         public boolean isValueEquality() {
-            return comparator.equals(Query.Comparator.EQV) && !(this instanceof Comparison.Variable);
+            return comparator.equals(Query.Comparator.EQV) && innerStatement() == null;
```

Comment:
Although this is "programmatically" correct, it hides the explicit meaning of the method which makes it one step harder to understand. For example:

When a reader reads this method, and ask: When is a ValueExecutor `isValueEquality`?
The answer, in this case, would be: when it holds a Value Equality comparator, and the `innerStatements()` is NULL.

However, the reader will still be left with a second/nested question: Why does having NULL `innerStatements()` make a ValueExecutor isValueEquality? The answer, in this case, would be: because when the `innerStatements()` is NULL, it means that this ValueExecutor is an instance of Variable Comparison. However, having NULL is an implementation detail.

Not only that the understanding of the method requires a dependency (thus adding "complexity"), it's also not the canonical meaning, and also not guaranteed to be true all the time. What if a different subclass of ValueExecutor implements innerStatements() to not be NULL, and still be `isValueEquality`? How would that affect the `isValueEquality` implementation? It would have to change too.

The change, in this case, is obviously very minor, but I believe this is a coding principle we should apply to improve the understandability of our code base. That is: to define the behaviour of a class/method as canonical as possible, which simplifies the understanding, and have it not be affected by the change of other non-related classes/methods.

As I start to understand more of our low-level implementations, I will highlight other places where this occurs. For now, I would recommend changing this method to the previous implementation, unless it would programmatically wrong/error.